### DISCUSSION TITLE
a best way get bitmap in recyclerview

### DISCUSSION THREAD

**Discussion by mikaelzero** [coil-kt/coil#878] [Q&A]
val request = ImageRequest.Builder(context)
            .data(maskImageUrl)
            .target(
                onSuccess = {
                    if (maskLayout.tag == maskImageUrl) {
                        maskLayout.changeBitmap(it.toBitmap())
                    }
                }
            )
            .build()
        context.imageLoader.enqueue(request)

If I use this way, the picture of the list will be messed up

**Comment by mikaelzero** [comment]
i tried this way


        CoilUtils.clear(materialMaskLayout)
        val materialRequest = ImageRequest.Builder(context)
            .data(maskImageUrl)
            .target(object : PoolableViewTarget<View> {
                override val view: View
                    get() = materialMaskLayout

                override fun onSuccess(result: Drawable) {
                    materialMaskLayout.changeBitmap(result.toBitmap())
                }

                override fun onClear() {
                    materialMaskLayout.changeBitmap(null)
                }

            })
            .build()
        context.imageLoader.enqueue(materialRequest)

**Reply by colinrtwhite** [reply]
It's not clear what's being asked here. Please add more info.
